cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -219,6 +226,13 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
     // In dataproc bucket, the shared folder for artifacts will be <bucket>/cdap-job/cached-artifacts.
     // All instances of CacheableLocalFile will be copied to the shared folder if they do not exist.
     String cacheRootPath = getPath(DataprocUtils.CDAP_GCS_ROOT, DataprocUtils.CDAP_CACHED_ARTIFACTS);
+    boolean enableGCSCachingSnapshot = Boolean.parseBoolean(
+      provisionerContext.getProperties().getOrDefault(DataprocUtils.ENABLE_GCS_CACHING_SNAPSHOT, "false"));
+    // cdap_version is of the form "major.minor.fix-buildTime"/"major.minor.fix-SNAPSHOT-buildTime"
+    String[] appCDAPVersion = cdapVersionInfo.toString().split("-");
+    boolean enableCommonArtifactsCaching = !disableGCSCaching &&
+      (!"SNAPSHOT".equals(appCDAPVersion[1]) || enableGCSCachingSnapshot);
```

Comment:
Check for the case in which array size is 1, just in case format changes in the future

Reply:
ack